src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -998,27 +1000,34 @@ vector<TripleWithPropertyPath> Visitor::visit(
   // Similarly if a triple `?var ql:contains-word "words"` is contained in the
   // query, then the variable `ql_matchingword_var` is implicitly created and
   // visible in the query body.
-  auto setMatchingWordAndTextscoreVisibleIfPresent =
-      [this](VarOrTerm& subject, VarOrPath& predicate, VarOrTerm& object) {
-        if (auto* var = std::get_if<Variable>(&subject)) {
-          if (auto* propertyPath = std::get_if<PropertyPath>(&predicate)) {
-            if (propertyPath->asString() == CONTAINS_WORD_PREDICATE) {
-              addVisibleVariable(var->getTextScoreVariable());
-              string name = object.toSparql();
-              for (std::string s : std::vector<std::string>(
-                       absl::StrSplit(name.substr(1, name.size() - 2), ' '))) {
-                if (s.back() != '*') {
-                  continue;
-                }
-                s.pop_back();
-                addVisibleVariable(var->getMatchingWordVariable(s));
-              }
-            } else if (propertyPath->asString() == CONTAINS_ENTITY_PREDICATE) {
-              addVisibleVariable(var->getTextScoreVariable());
+  auto setMatchingWordAndTextscoreVisibleIfPresent = [this](
+                                                         VarOrTerm& subject,
+                                                         VarOrPath& predicate,
+                                                         VarOrTerm& object) {
+    if (auto* var = std::get_if<Variable>(&subject)) {
+      if (auto* propertyPath = std::get_if<PropertyPath>(&predicate)) {
+        if (propertyPath->asString() == CONTAINS_WORD_PREDICATE) {
+          addVisibleVariable(var->getTextScoreVariable());
+          string name = object.toSparql();
+          if (!((name[0] == '"' && name[name.size() - 1] == '"') ||
+                (name[0] == '\'' && name[name.size() - 1] == '\''))) {
+            AD_THROW(
+                "ql:contains-word has to be followed by a string in quotes");
+          }
```

Comment:
Use the errorHandlingMechanism that the parser also uses otherwise (I think it is used `reportError`, that way we get nicer feedback in the UI.